src/jit/codegenxarch.cpp

```diff
@@ -6827,6 +6860,12 @@ void        CodeGen::genCompareLong(GenTreePtr  treeNode)
     assert(varTypeIsLong(op1->TypeGet()) && varTypeIsLong(op2->TypeGet()));
     regNumber targetReg  = treeNode->gtRegNum;
 
+    if (targetReg == REG_NA)
```

Comment:
Why incur function call overhead for this case?  How about genCodeForTreeNode() does the following for 32-bit while generating code for compare operators?

```
#if !defined(_TARGET_64BIT_)
            // X86 Long comparison
            else if (varTypeIsLong(op1Type))
            {
                **if ( (treeNode->gtRegNum != REG_NA)**
                    genCompareLong(treeNode);
            }
#endif // !defined(_TARGET_64BIT_)
```

After that we can assert here that targetReg is a valid reg